datafusion/core/src/physical_plan/planner.rs

```diff
@@ -1773,7 +1771,7 @@ impl DefaultPhysicalPlanner {
     where
         F: FnMut(&dyn ExecutionPlan, &dyn PhysicalOptimizerRule),
     {
-        let optimizers = &session_state.physical_optimizers;
+        let optimizers = session_state.physical_optimizers();
```

Comment:
This is quite a peculiar API, I'm not sure why the physical planner would be calling the optimizers and not SessionState itself

Reply:
it may predate when the list of optimizers was on the session state 🤔

Reply:
Ok - I will add it to my list